rust/tests/test_write_resiliency.rs

```diff
@@ -68,79 +72,104 @@ mod test {
     }
 
     async fn test_write_failures() -> Result<()> {
```

Comment:
Can you add a test for 3 total write failures to show that a datanode definitely had to be added for that to succeed?

Reply:
done

Reply:
make sense.